### PR TITLE
Rework delete service rules API to use RuleManager implementation

### PR DESCRIPTION
**Pull request by rozdy**

### CODE REVIEW

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.openkilda.floodlight.api.request.rulemanager.OfCommand;
+import org.openkilda.model.SwitchId;
+
+import lombok.Getter;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public class CommandsQueue {
```
Comment (sergii-iakovenko) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java:27]: Please, add a javadoc to the class.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public boolean hasNext() {
+        return iterator.hasNext();
+    }
+
+    public CommandsBatch getNext() {
+        return iterator.next();
+    }
+
+    public List<OfCommand> getFirst() {
+        return commandsBatches.size() > 0 ? commandsBatches.get(0).getCommands() : Collections.emptyList();
```
Comment (sergii-iakovenko) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java:51]: Why not use LinkedList for commandsBatches?

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java [hunk 3]
```diff
@@ -405,11 +471,24 @@ private void rulesResponse(String key, SwitchRulesResponse response) {
         }
     }
 
-    private void rulesResponse(String key, SpeakerCommandResponse response) {
+    private void rulesResponse(String key, SpeakerCommandResponse response) throws UnexpectedInputException {
```
Comment (sergii-iakovenko) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java:474]: Please, revise the method name.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java [hunk 4]
```diff
@@ -325,6 +331,89 @@ private Predicate<SpeakerData> buildPredicate(InstallRulesAction action) {
         }
     }
 
+    private boolean isInstallRequired(DeleteRulesAction action) {
+        return action == DROP_ALL_ADD_DEFAULTS || action == REMOVE_ADD_DEFAULTS || action == OVERWRITE_DEFAULTS;
+    }
+
+    private Predicate<SpeakerData> buildDeletePredicate(DeleteRulesAction action) {
```
Comment (sergii-iakovenko) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java:338]: Does it make sense to extract this code into a dedicated builder class?

**Review by sergii-iakovenko — COMMENTED** [review]
LGTM

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java (hunk 1 above)
Comment (rozdy) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java:30]: Added.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java (hunk 2 above)
Comment (rozdy) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/CommandsQueue.java:54]: I don't see a profit to use LinkedList. This one line will be slightly better, but lines where we create CommandsQueue will be slightly worse.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java (hunk 3 above)
Comment (rozdy) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java]: Done.

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java (hunk 4 above)
Comment (rozdy) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java]: Moved to separate class.

**Review by dpoltavets — APPROVED** [review]

**Comment by rozdy** [comment]
NOTE: merge after #4804

**Pull request merged**